aerospike/aerospike-client-python
# [CLIENT-2258] Backport 6.*: Remove auto-serialization and auto-deserialization

Docs: https://aerospike-python-client--488.org.readthedocs.build/en/488/data_mapping.html

Build wheels all passes: https://github.com/aerospike/aerospike-client-python/actions/runs/5896059798

Proof it works:

```python
jnguyen@as-PF3WN8Y6:~/aerospike/aerospike-client-python$ python3
Python 3.9.16 (main, Feb  8 2023, 16:50:17) 
[GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aerospike
>>> aerospike.__version__
'12.0.0'
>>> config = { 'hosts': [ ('127.0.0.1',3000)]}
>>> client = aerospike.client(config).connect()
>>> client.put(("test", "demo", 2), {"a": (1, 2, 3)})
0
>>> exit()
jnguyen@as-PF3WN8Y6:~/aerospike/aerospike-client-python$ python3 -m pip install aerospike-6.2.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl 
...
jnguyen@as-PF3WN8Y6:~/aerospike/aerospike-client-python$ python3
Python 3.9.16 (main, Feb  8 2023, 16:50:17) 
[GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aerospike
>>> config = { 'hosts': [ ('127.0.0.1',3000)]}
>>> client = aerospike.client(config).connect()
>>> _, _, bins = client.get(("test", "demo", 2))
>>> bins["a"]
bytearray(b'\x80\x04\x95\t\x00\x00\x00\x00\x00\x00\x00K\x01K\x02K\x03\x87\x94.')
>>> import pickle
>>> pickle.loads(bins["a"])
(1, 2, 3)
```

* Return a client error when writing unsupported Python types to the server
* Convert AS_BYTES_PYTHON server types to bytearrays when reading data from the server
* Remove support for serializing booleans as the AS_BYTES_PYTHON server type
* Remove aerospike.SERIALIZER_PYTHON
* client.put(): set serializer parameter's default value to aerospike.SERIALIZER_NONE
* Add extra tests to verify client.operate() and expression behavior with instance-level serializers and deserializers
* client.put(): fix bug where Python bytes bin values can't be used if serializer parameter is set to SERIALIZER_NONE
* Docs: data mapping: add headers
* Add build wheels workflow
* Change send_bool_as default value to AS_BOOL
* Build custom version of C client that is compatible with OpenSSL 1.1 or later
